### PR TITLE
br: add retry for azblob read file

### LINKED ISSUE #39649: azblob need retry when connection error occurs
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
run br to restore 

### 2. What did you expect to see? (Required)
success
### 3. What did you see instead (Required)
```
[INFO] [collector.go:69] ["restore log failed summary"] [error="scanning metadata meets error Failed to list azure blobs, bucket(container)='backup-data': context canceled: during reading meta file v1/backupmeta/437743859352207656-bef2cb55-5563-4a18-b00e-208d442763a2.meta from storage: Failed to read azure blob file, file info: bucket(container)='backup-data', key='log-adhoc/v1/backupmeta/437743859352207656-bef2cb55-5563-4a18-b00e-208d442763a2.meta': read tcp [172.*.*.*:*](http://172.*.*.*:*/)->[20.*.*.*:443](http://20.*.*.*:443/): read: connection reset by peer"]
```

### PR DESCRIPTION
**Pull request by Leavrth**
Signed-off-by: Leavrth <jianjun.liao@outlook.com>

### What problem does this PR solve?

Issue Number: close #39649 

Problem Summary:
azblob need retry when connection error occurs

### What is changed and how it works?
retry to read file when the error is retryable.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/storage/azblob.go [hunk 1]
```diff
@@ -280,10 +283,23 @@ func (s *AzureBlobStorage) WriteFile(ctx context.Context, name string, data []by
...
+	var err error
+	for retryTimes := 0; ; retryTimes++ {
+		if retryTimes == maxRetry {
+			return nil, errors.Annotatef(err, "Failed to retry to download azure blob file, file info: bucket(container)='%s', key='%s'", s.options.Bucket, s.withPrefix(name))
+		}
+		resp, err = client.Download(ctx, nil)
+		if err != nil {
+			if utils.MessageIsRetryableStorageError(err.Error()) {
+				log.Warn("Failed to download azure blob file, file info", zap.String("bucket(container)", s.options.Bucket), zap.String("key", s.withPrefix(name)), zap.Int("retry", retryTimes), zap.Error(err))
+				continue
```
Comment (3pointer) [inline br/pkg/storage/azblob.go:296]: shall we sleep for a while?

Comment (3pointer) [reply br/pkg/storage/azblob.go:296]: can we use `utils.WithRetry`?

File: br/cmd/br/cmd.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
 	"github.com/pingcap/tidb/br/pkg/gluetidb"
+	"github.com/pingcap/tidb/br/pkg/metautil"
```
Comment (hawkingrei) [inline br/cmd/br/cmd.go:17]: Please run ```make bazel_prepare``` to upgrade the bazel config.

**Comment by joccau** [comment]
Can we have any test ?

**Comment by Leavrth** [comment]
> Can we have any test ?

Currently manually set breakpoint to test to prove that it will retry if failed.

**Review by joccau — APPROVED** [review]
LGTM

**Pull request merged**